src/topology/mod.rs

```diff
@@ -1056,60 +1229,103 @@ mod tests {
     /// Generate valid (disjoint) roots for [`Topology::distribute_items()`],
     /// taken from [`Topology::test_instance()`]
     fn disjoint_roots() -> impl Strategy<Value = Vec<&'static TopologyObject>> {
```

Comment:
I discovered that the existing disjoint_roots impl was bad (favored specific layers of the hierarchy) while using it to test group creation, so I fixed it.